solutions/cses-1650.mdx

```diff
@@ -2,41 +2,116 @@
 id: cses-1650
 source: CSES
 title: Range Xor Queries
-author: Danh Ta Chi Thanh
+author: Danh Ta Chi Thanh, Nathan Gong
 ---
 
-## Solution
+## Explanation
 
-[Unofficial Editorial](https://codeforces.com/blog/entry/77128?#comment-637536)
+Whenever we come across range queries, prefix sums should immediately come to
+mind. However, since we are dealing with range XOR, we have to use a prefix XOR
+array rather than a prefix sum array. To do so, we can define $\texttt{prefix}$
+as an array such that
+$\texttt{prefix}[i]= \texttt{arr}[0] \oplus \texttt{arr}[1] \oplus \dots \oplus \texttt{arr}[i]$.
+Thus, for each query $(a, b)$, our answer is simply
+$\texttt{prefix}[a-1]\oplus \texttt{prefix}[b]$.
+
+### Proof
+
+We know that
+$\texttt{prefix}[a-1]= \texttt{arr}[0] \oplus \texttt{arr}[1] \oplus \dots \oplus \texttt{arr}[a-1]$.
+We also know that
+$\texttt{prefix}[b]= \texttt{arr}[0] \oplus \texttt{arr}[1] \oplus \dots \oplus \texttt{arr}[b]$.
+
+Therefore:
+
+$\texttt{prefix}[a-1] \oplus \texttt{prefix}[b]= \underbrace{\texttt{arr}[0] \oplus \texttt{arr}[0] \oplus \texttt{arr}[1] \oplus \texttt{arr}[1] \oplus \dots \oplus \texttt{arr}[a-1] \oplus \texttt{arr}[a-1]}_{\text{cancels out to 0}} \oplus \texttt{arr}[a] \oplus \texttt{arr}[a+1] \dots \oplus \texttt{arr}[b]$.
+
+This then simplifies to
+$\texttt{arr}[a] \oplus \texttt{arr}[a+1] \dots \oplus \texttt{arr}[b]$, which
+is exactly what we're looking for.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(n+q)$
+
+<LanguageSection>
+
+<CPPSection>
 
 ```cpp
 #include <bits/stdc++.h>
 
 using namespace std;
 
-//constant initialization
-const int maxn=2e5+10;
+const int maxn = 2e5 + 10;
 
-//variables used for the current problem
-long long n,q,a[maxn],u,v;
+int n, q, arr[maxn], prefix[maxn], a, b;
 
-void solve(){
+void solve() {
 	cin >> n >> q;
```

Comment:
wait why not just put this in main there's literally no reason for it to be in a separate method